### PR TITLE
Feature/poll for new queues

### LINKED ISSUE #2: Repair worker poll for new queues

### PR DESCRIPTION
**Pull request by devshorts**
Fixes #2 (with a very basic implementation)

Repair worker manager now polls for queues to monitor.

Lots of issues with this in terms of concurrent work and scaleability. I will be addressing a very SIMPLE version of concurrent limitation using leadership election in another branch and we should refactor the repair manager completely to be both leader elected AND fully async (insetad of blocking on a scheduled pool right now)

### CODE REVIEW

File: core/src/main/java/io/paradoxical/cassieq/bundles/GuiceBundleProvider.java [hunk 1]
```diff
@@ -31,12 +35,11 @@ public GuiceBundleProvider() {
         final GuiceBundle.Builder<ServiceConfiguration> builder = GuiceBundle.<ServiceConfiguration>newBuilder();
 
         builder.enableAutoConfig(ServiceApplication.class.getPackage().getName())
-               .setConfigClass(ServiceConfiguration.class);
+               .setConfigClass(ServiceConfiguration.class)
+               .setInjectorFactory((stage, modules) -> Governator.createInjector(modules));
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/bundles/GuiceBundleProvider.java:39]: Tyte

File: core/src/main/java/io/paradoxical/cassieq/model/time/Clock.java [hunk 2]
```diff
@@ -7,5 +7,7 @@
     Instant now();
 
     void sleepFor(Duration duration) throws InterruptedException;
+
+    long jitter(int i);
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/model/time/Clock.java:11]: we need to improve this parameter name. i is unclear.

File: core/src/main/java/io/paradoxical/cassieq/workers/repair/RepairWorkerImpl.java [hunk 3]
```diff
@@ -213,7 +208,7 @@ private void waitForTimeout(final DateTime tombstoneTime) {
     }
 
     private void deleteMessagesInBucket(final RepairBucketPointer currentBucket) {
-        if (configuration.isDeleteBucketsAfterRepair()) {
+        if (queueDefinition.getDeleteBucketsAfterFinaliziation()) {
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/workers/repair/RepairWorkerImpl.java:211]: too bad we can't all this `shouldDeleteBucketsAfterFinalization`

File: core/src/main/java/io/paradoxical/cassieq/model/QueueDefinition.java [hunk 4]
```diff
@@ -13,22 +13,31 @@
     private final Integer maxDeliveryCount;
     private final QueueStatus status;
     private final int version;
+    private final Integer repairWorkerPollFrequencySeconds;
+    private final Integer repairWorkerTombstonedBucketTimeoutSeconds;
+    private final Boolean deleteBucketsAfterFinaliziation;
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/model/QueueDefinition.java:18]: finalization is spelled wrong

File: core/src/main/java/io/paradoxical/cassieq/workers/repair/SimpleRepairWorkerManager.java [hunk 5]
```diff
@@ -20,56 +25,104 @@
...
+    @Override
+    public synchronized void notifyChanges() {
+        try {
+            final Set<RepairWorkerKey> expectedWorkers = getExpectedWorkers();
+
+            final ImmutableSet<RepairWorkerKey> newWorkers = Sets.difference(expectedWorkers, currentRepairWorkers).immutableCopy();
+
+            final ImmutableSet<RepairWorkerKey> itemsToStop = Sets.difference(currentRepairWorkers, expectedWorkers).immutableCopy();
+
+            logger.with("count", itemsToStop.size()).info("Removing workers");
```
Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/workers/repair/SimpleRepairWorkerManager.java:96]: cool! very simple and straight forward. love it

**Comment by jakeswenson** [comment]
:heavy_check_mark: LGTM

File: core/src/main/java/io/paradoxical/cassieq/workers/repair/RepairWorkerImpl.java (hunk 3 above)
Comment (devshorts) [inline core/src/main/java/io/paradoxical/cassieq/workers/repair/RepairWorkerImpl.java:211]: I mean we couuuuld, we just like lombok a lot though :palm_tree:

Comment (jakeswenson) [inline core/src/main/java/io/paradoxical/cassieq/workers/repair/RepairWorkerImpl.java:211]: can you change the prefix with `@Getter`?

**Pull request merged**